core/src/main/scala/org/bitcoins/core/api/dlcoracle/OracleEvent.scala

```diff
@@ -130,10 +135,29 @@ case class CompletedDigitDecompositionV0OracleEvent(
     maturationTime: Instant,
     announcementSignature: SchnorrDigitalSignature,
     eventDescriptorTLV: DigitDecompositionEventDescriptorV0TLV,
-    outcomes: Vector[DLCAttestationType],
+    outcomes: Vector[DigitDecompositionAttestationType],
```

Comment:
Why don't we just combine these two fields into one type? It seems that should be used together no? 

What I had in mind what something like 

```scala
case class NumericDLCOutcome(outcomes: Vector[DigitDecompositionAttestationType]) extends DLCOutcome { 
   val attesations: Vector[FieldElement] = outcome.map(_.fieldElement) // or something like this
 }
```

and then we can just have 

```scala
case class CompletedDigitDecompositionV0OracleEvent(
    pubkey: SchnorrPublicKey,
    nonces: Vector[SchnorrNonce],
    eventName: String,
    signingVersion: SigningVersion,
    maturationTime: Instant,
    announcementSignature: SchnorrDigitalSignature,
    eventDescriptorTLV: DigitDecompositionEventDescriptorV0TLV
    outcome: NumericDLCOutcome) { 
       val attestations: Vector[FieldElement] = outcome.attestations
      val outcomes: Vector[DigitDecompositionAttestationType] = outcome.outcomes
    }
```